### PR TITLE
Add golangci-lint

### PR DESCRIPTION
**Pull request by chris124567**
Add golangci-lint and fix a couple issues it found.  I used the config file currently in hostd.  I think the linters are a bit nitpicky (they say having `banned` and `Banned` on EphemeralPeerStore is confusing) but I didn't disagree enough to make an exception with `//nolint`.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      - equalFold
+  revive:
+    ignore-generated-header: true
+    rules:
+      - name: blank-imports
+        disabled: false
+      - name: bool-literal-in-expr
+        disabled: false
+      - name: confusing-naming
+        disabled: false
```
Comment (lukechampine) [inline .golangci.yml:93]: is this the check triggered by `banned`? I'd rather just disable it.

File: wallet/seed.go [hunk 2]
```diff
@@ -128,14 +128,16 @@ outer:
...
+				}
+				index, ok := sav.addrs[addr]
+				if !ok {
 					return fmt.Errorf("missing key for ID %v", parent)
-				} else {
-					SignTransaction(cs, txn, sigIndex, sav.seed.PrivateKey(index))
-					continue outer
 				}
+				SignTransaction(cs, txn, sigIndex, sav.seed.PrivateKey(index))
+				continue outer
```
Comment (lukechampine) [inline wallet/seed.go:140]: I understand the rationale behind this check, but I still resent it. It directly conflicts with `if`-scoped declarations, which imo are often preferable

Comment (chris124567) [reply wallet/seed.go:140]: I didn't like this either.  Should we just disable the check here?

Comment (lukechampine) [reply wallet/seed.go:140]: can we disable it universally?

Comment (chris124567) [reply wallet/seed.go:140]: Yeah in .golangci.yml -  I'll do it.

**Comment by n8maninger** [comment]
Snuck this in in #44

**Pull request closed without merging**